core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooLock.java

```diff
@@ -286,7 +286,7 @@ public void process(WatchedEvent event) {
                   else if (asyncLock != null)
                     failedToAcquireLock();
                 }
-              } catch (Throwable e) {
+              } catch (KeeperException | InterruptedException e) {
```

Comment:
I don't think this catch should be narrowed because any unexpected runtime exception needs to be properly handled for zookeeper locks.  We don't want to ignore an NPE and think the lock is held when it is not.

I worry about the correctness of this change because of possible issues like this.  Narrowing could lead to handling runtime exceptions differently which in some case could cause bugs.  Its hard to review all of the narrowing for correctness.

Reply:
Could just do `(KeeperException | InterruptedException | RuntimeException e)`. I still think there's merit in no longer catching `Error` and thinking we have any clue how to handle that case (especially with merely logging it).

Reply:
This code does more than log, it calls `lockWatcher.unableToMonitorLockNode(e)` letting an observer know that the state of the lock is now unknown.  With ZK lock issues we do want to ensure the process halts.  It may make sense to catch Throwable in case, I don't know without further inspection of the code.